sklearn_numba_dpex/common/_utils.py

```diff
@@ -88,3 +89,25 @@ def _check_max_work_group_size(
         )
     else:
         return work_group_size
+
+
+_GLOBAL_MEM_CACHE_SIZE_DEFAULT = 1048576
```

Comment:
Is it possible to document the choice of this value?

Also I think it is worth making the power of 2 explicit
```suggestion
_GLOBAL_MEM_CACHE_SIZE_DEFAULT = 1048576  # 2**20
```

Or even this is clearer to me:

```suggestion
_GLOBAL_MEM_CACHE_SIZE_DEFAULT = 2**20
```

What do you think?